### DISCUSSION TITLE
How to setup custom module paths for ZLS with no `build.zig`?

### DISCUSSION THREAD

**Discussion by heysokam** [zigtools/zls#1981] [Q&A]
I'm writing a custom buildsystem tool, so relying on `build.zig` functionality does not solve the problem.
So I was wondering if there is a way to explicitly specify paths where zls will search for modules :thinking:

**Comment by techatrix** [comment]
Without knowing anything about the custom buildsystem tool I can't give a clear answer but how having your buildsystem tool generate a `build.zig` file? The `build.zig` doesn't need to be fully functional, it only need to declare necessary modules.

This example should be enough for ZLS to find some modules:
```zig
const std = @import("std");
pub fn build(b: *std.Build) !void {
    const target = b.standardTargetOptions(.{});
    const optimize = b.standardOptimizeOption(.{});

    const mod1 = b.addModule("mod1", .{ .root_source_file = b.path("<path>") });
    const mod2 = b.addModule("mod2", .{ .root_source_file = b.path("<path>") });
    const mod3 = b.addModule("mod3", .{ .root_source_file = b.path("<path>") });

    const exe = b.addExecutable(.{
        .name = "fake",
        .root_source_file = b.path("<path>"),
        .target = target,
        .optimize = optimize,
    });
    b.installArtifact(exe);
    exe.root_module.addImport("<name>", mod1);
    exe.root_module.addImport("<name>", mod2);
    exe.root_module.addImport("<name>", mod3);
}
```

**Reply by heysokam** [reply]
That's definitely an idea.
But my question was more about a configuration option/file,
where I could send a `-Mname=path/to/file.zig` command so that the LSP has what it needs
without need for custom code+file generation, which creates a whole lot of problems when dealing with any form of non-orthodox file/folder structure.
